packages/aws-rfdk/lib/deadline/lib/sep-configuration.ts

```diff
@@ -0,0 +1,403 @@
+}
+
+/**
+ * Different logging levels.
+ */
+export enum SpotEventPluginLoggingLevel {
+  /**
+   * Standard logging level.
+   */
+  STANDARD = 'Standard',
+
+  /**
+   * Detailed logging about the inner workings of the Spot Event Plugin.
+   */
+  VERBOSE = 'Verbose',
+
+  /**
+   * All Verbose logs plus additional information on AWS API calls that are used.
+   */
+  DEBUG = 'Debug',
+
+  /**
+   * No logging enabled.
+   */
+  OFF = 'Off',
+}
+
+/**
+ * How the Spot Event Plugin should handle Pre Job Tasks.
+ */
+export enum SpotEventPluginPreJobTaskMode {
+  /**
+   * Only start 1 Spot instance for the pre job task and ignore any other tasks for that job until the pre job task is completed.
+   */
+  CONSERVATIVE = 'Conservative',
+
+  /**
+   * Do not take the pre job task into account when calculating target capacity.
+   */
+  IGNORE = 'Ignore',
+
+  /**
+   * Treat the pre job task like a regular job queued task.
+   */
+  NORMAL = 'Normal',
+}
+
+/**
+ * The Worker Extra Info column to be used to display AWS Instance Status
+ * if the instance has been marked to be stopped or terminated by EC2 or Spot Event Plugin.
+ * See "AWS Instance Status" option at https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/event-spot.html?highlight=spot%20even%20plugin#event-plugin-configuration-options
+ */
+export enum SpotEventPluginAwsInstanceStatus {
+  DISABLED = 'Disabled',
+  EXTRA_INOF_0 = 'ExtraInfo0',
+  EXTRA_INOF_1 = 'ExtraInfo0',
+  EXTRA_INOF_2 = 'ExtraInfo0',
+  EXTRA_INOF_3 = 'ExtraInfo0',
+  EXTRA_INOF_4 = 'ExtraInfo0',
+  EXTRA_INOF_5 = 'ExtraInfo0',
+  EXTRA_INOF_6 = 'ExtraInfo0',
+  EXTRA_INOF_7 = 'ExtraInfo0',
+  EXTRA_INOF_8 = 'ExtraInfo0',
+  EXTRA_INOF_9 = 'ExtraInfo0',
+}
+
+/**
+ * Spot Event Plugin configuration options
+ */
+export interface SEPConfigurationProperties {
+  /**
+   * The array of Spot Event Plugin spot fleets used to generate the mapping between your groups and spot fleet requests.
+   *
+   * @default Spot Fleet Request Configurations will not be updated.
+   */
+  readonly spotFleets?: SEPSpotFleet[];
+
+  /**
+   * How the event plug-in should respond to events.
+   *
+   * @default SpotEventPluginState.DISABLED
+   */
+  readonly state?: SpotEventPluginState;
+
+  /**
+   * Determines whether Deadline Resource Tracker should be used.
+   * Only disable for AMIs with Deadline 10.0.26 or earlier.
+   *
+   * @default true
+   */
+  readonly enableResourceTracker?: boolean;
+
+  /**
+   * Spot Event Plugin logging level.
+   *
+   * @default SpotEventPluginLoggingLevel.STANDARD
+   */
+  readonly loggingLevel?: SpotEventPluginLoggingLevel;
+
+  /**
+   * The AWS region in which to start the spot fleet request.
+   *
+   * @default The region of the current stack.
+   */
+  readonly region?: string;
+
+  /**
+   * Number of minutes that a AWS Worker will wait in a non-rendering state before it is shutdown.
+   *
+   * @default 10
+   */
+  readonly idleShutdown?: number;
```

Comment:
I was thinking about it, but Spot Event plugin expects an integer meaning the number of minutes here, so customers might use `Duration.seconds(30)` and this would result in undefined behavior.

So I'm afraid this might be more confusing to customers than beneficial. What do you think?

Reply:
Cool! I didn't check `(opts)` before.

Reply:
Looks good! thanks